From 30a0c40f49a6df654d153eb0ff1deb5e1fbd7b4f Mon Sep 17 00:00:00 2001
From: Sandor Yu <R01008@freescale.com>
Date: Thu, 14 May 2015 11:24:36 +0800
Subject: [PATCH 492/505] MLK-10876: DCIC: System hang when run dcic unit test
 if fb blank

DCIC working depend on framebuffer driver,
if fb in blank state and fb no pixel clock output to dcic,
dcic will hang.
So add fb unblank function at the begin of test application
to fix the issue.

Signed-off-by: Sandor Yu <R01008@freescale.com>
---
 test/mxc_dcic_test/mxc_dcic_test.c | 5 ++++-
 1 file changed, 4 insertions(+), 1 deletion(-)

diff --git a/test/mxc_dcic_test/mxc_dcic_test.c b/test/mxc_dcic_test/mxc_dcic_test.c
index fbe593d..f1c5b4a 100644
--- a/test/mxc_dcic_test/mxc_dcic_test.c
+++ b/test/mxc_dcic_test/mxc_dcic_test.c
@@ -1,5 +1,5 @@
 /*
- * Copyright (C) 2014 Freescale Semiconductor, Inc. All rights reserved.
+ * Copyright (C) 2014-2015 Freescale Semiconductor, Inc. All rights reserved.
  */
 
 /*
@@ -452,6 +452,9 @@ int main(int argc, char **argv)
 		printf("open /dev/dcic1\n");
 	}
 
+	/* unblank framebuffer */
+	ioctl(fd_fb0, FBIOBLANK, FB_BLANK_UNBLANK);
+
 	retval = ioctl(fd_fb0, FBIOGET_VSCREENINFO, &screen_info);
 	if (retval < 0) {
 		goto err2;
-- 
1.8.0

